File: ksum.py

l = [2,3,4,9]
k = 6




def s(arr,i,k):
    temp = []
    for j in arr:
        val = j / i
        t = isInt(val)
        if t :
            temp.append(int(val))
        else:
            temp.append(int(val) + 1)

        # temp.append()
    if sum(temp) <=k:
        # print(temp)
        return True
    else:
        return False

def isInt(val):
    if val % 1 == 0:
        return True
    else :
        return False

for i in range(1,max(l)):
    x = s(l,i,k)
    if x:
        print(i)
        break

File: ksum.java


import java.util.*;
class Main{
  public static void main(String[] args){
    Scanner input = new Scanner(System.in);
    int size = input.nextInt();
    int k = input.nextInt();
    int [] arr = new int[size];
    for(int i = 0; i < size; i++){
      arr[i] = input.nextInt();
    }
    int max = 0;
    for (int i = 0; i < size; i++){
      if(arr[i] > max){
        max = arr[i];
      }
    }
    for (int i = 1; i < max; i++){
      int mid  = i;
      long sum = 0;
      for(int j = 0; j < size; j++){
        sum = sum + (arr[j] + mid - 1)/mid;
      }
      if(sum <= k){
        System.out.print(i);
        break;
      }
    }
  }
}
